tests/func/test_listing.py

```diff
@@ -0,0 +1,34 @@
+from datachain.client.local import FileClient
+from datachain.lib.dc import DataChain
+from datachain.lib.listing import list_bucket
+from tests.data import ENTRIES
+
+
+def test_listing_generator(cloud_test_catalog, cloud_type):
+    ctc = cloud_test_catalog
+
+    uri = f"{ctc.src_uri}/cats"
+
+    dc = DataChain.from_records(DataChain.DEFAULT_FILE_RECORD).gen(
```

Comment:
How can we avoid this DEFAULT_FILE_RECORD? 🤔

Reply:
The problem is that generator generates new rows for each row in the chain so if we have empty chain no rows will be generated. If we want some function to run on empty chain, then we would need to re-design some parts of internal generator logic I think. This needs to be investigated but I would do it in separate issue / PR as it could be big change by itself.